QuadrupedalRobots/ETGRL/deployment/envs/EnvWrapper.py

```diff
@@ -0,0 +1,336 @@
+            obs_list = []
+            for t in range(self.time_steps):
+                obs_list.append(copy(self.obs_history[t*self.time_interval]))
+            obs_list.append(copy(obs))
+            self.obs_history[-1] = copy(obs)
+            if self.sensor_mode["RNN"]["mode"]=="GRU":
+                obs = np.stack(obs_list,axis=0)
+            elif self.sensor_mode["RNN"]["mode"]=="stack":
+                obs = np.array(obs_list).reshape(-1)
+        return obs,info
+    
+    def step(self,action,**kwargs):
+        self.env.step(action,**kwargs)
+
+class ControlLoopWrapper(object):
+    def __init__(self,env,dt,action_repeat):
+        self.env = env
+        self.dt = dt
+        self.t0 = 0
+        self.obs_dim = self.env.get_obs_dim()
+
+    def get_obs_dim(self):
+        return self.obs_dim
+
+    def reset(self,**kwargs):
+        obs,info = self.env.reset(**kwargs)
+        self.t0 = time.clock()
+        return obs,info
+    
+    def get_time(self):
+        return time.clock()-self.t0
+    
+    def step(self,action,**kwargs):
+        t_start = time.clock()
+        obs,info = self.env.step(action,**kwargs)
+        t_now = time.clock()-t_start
+        print("t_now:",t_now)
+        if t_now<self.dt-5e-4:
+            time.sleep(self.dt-t_now)
+        return obs,info
+
+
+class ActionFilterWrapper(object):
+    def __init__(self,env,robot,dt,enable_action_filter):
+        self.env = env
+        self.dt = dt
+        self.robot = robot
+        self.enable_action_filter = enable_action_filter
+        if self.enable_action_filter:
+            self._action_filter = self._BuildActionFilter()
+
+    def reset(self,**kwargs):
+        obs_all,info = self.env.reset(**kwargs)
+        self._step_counter = 0
+        if self.enable_action_filter:
+            self._ResetActionFilter()
+        return obs_all,info
+    
+    def step(self,action,**kwargs):
+        if self.enable_action_filter:
+            action = self._FilterAction(action)
+        self.env.step(action,**kwargs)
+        self._step_counter += 1
+        # return obs_all, info
+    
+    def get_observation(self):
+        return self.env.get_observation()
+
+
+    def _BuildActionFilter(self):
+        sampling_rate = 1 / self.dt
+        num_joints = 12
+        a_filter = action_filter.ActionFilterButter(sampling_rate=sampling_rate,
+                                                    num_joints=num_joints)
+        return a_filter
+
+    def _ResetActionFilter(self):
+        self._action_filter.reset()
+
+    def _FilterAction(self, action):
+        # initialize the filter history, since resetting the filter will fill
+        # the history with zeros and this can cause sudden movements at the start
+        # of each episode
+        if self._step_counter == 0:
+            default_action = np.array([0,0.9,-1.8]*4)
+            self._action_filter.init_history(default_action)
+            # for j in range(10):
```

Comment:
Pls remove these lines.

Reply:
> @HaojieSHI98
> 
> * Please remove the `__pycache__` directory.
> * Please remove the npy/pt files if they are not used.
> * It looks like we don't provide the code of the sim2real part.
> * For README.md, we'd better provide a tutorial to introduce how to train a deployable model from scratch.

For the npy/pt files, it's for demonstrate. Shall we remove them?